rfc/rfc-43/rfc-43.md

```diff
@@ -0,0 +1,316 @@
+
+    - | name         | type   | comment               |
+            | ------------ | ------ | --------------------- |
+      | id           | bigint | auto_increment        |
+      | db_name      | string | database name         |
+      | table_name   | string | table name            |
+      | base_path    | string | storage path          |
+      | owner        | string | owner                 |
+      | queue        | string | queue                 |
+      | action_types | string | register action types |
+      | priority     | string | priority              |
+
+- **operation**
+
+    - unique key: tbl_id, instant
+
+    - | name           | type      | comment                                            | 
+      | -------------- | --------- | -------------------------------------------------- |
+      | operation_id   | bigint    | auto_increment                                     |
+      | tbl_id         | string    | table the operation belongs to                     |
+      | instant        | string    | operation timestamp                                |
+      | action         | string    | operation action, compaction, clustering and etc   |
+      | status         | int       | operation status                                   |
+      | engine         | string    | operation engine, Spark or Flink                   |
+      | retry_on_error | boolean   | operation should retry on error or not             |
+      | run_times      | int       | operation run times                                |
+      | isDeleted      | boolean   | true if this operation has been deleted in service |
+      | schedule_time  | timestamp | operation created time                             |
+      | create_time    | timestamp | operation created time                             |
+      | update_time    | timestamp | operation updated time                             |
+
+## Interface design
+
+### Register
+
+/v1/hoodie/service/register
+
+```
+{
+"table_name":"table_name",
+"base_path":"/hoodie/base_path",
+"owner":"default",
+"queue":"root.default"
+"priority":"0"
+}
+```
+
+## Error Handling
+
+Due to two tables being involved in each operation, we need to ensure that they are in sync and errors during dataset
+operation are handled correctly.
+
+1. Client scheduled the plan but request failed.
+
+2. RequestHandler received request but the commit is not completed.
+
+3. Client rollback plan after request to Compaction/Clustering Service.
+
+Client will report to service all unCompleted plans by time or num deltaCommits or combination, the service upsert to
+meta table uses HoodieServicePayload to guarantee any plan unique. Client use Delete request to rollback plan for
+Compaction/Clustering Service.
+
+### Configuration
+
+hoodie.table.services.enable=true
+hoodie.table.service.compaction.enable=true
+hoodie.table.service.clustering.enable=true
+hoodie.table.service.clean.enable=true
+
+## Proposal
```

Comment:
this is not proposal section, this is implementation plan. The whole document is a proposal.